platform/pulp/platform/models/content.py

```diff
@@ -0,0 +1,94 @@
+"""
+Content related Django models.
+"""
+import hashlib
+
+from django.contrib.contenttypes import fields
+from django.db import models
+
+
+from pulp.platform.models import Model, MasterModel, Notes
+from pulp.platform.models.storage import StoragePath
+
+
+class Content(MasterModel):
+    """
+    A piece of managed content.
+
+    :cvar natural_key_fields: Tuple of natural fields.  Must be: models.Field.
+    :type natural_key_fields: tuple
+
+    Fields:
+
+    :cvar type: The content type.
+    :type type: models.TextField
+
+    Relations:
+
+    :cvar notes: Arbitrary information stored with the content.
+    :type notes: fields.GenericRelation
+    """
+    natural_key_fields = ()
+
+    type = models.TextField(blank=False, default=None)
+
+    notes = fields.GenericRelation(Notes)
+
+    def natural_key(self):
+        return (getattr(self, f.name) for f in self.natural_key_fields)
+
+    def natural_key_digest(self):
+        h = hashlib.sha256()
+        for name in sorted(f.name for f in self.natural_key_fields):
+            h.update(name)
+            value = getattr(self, name)
+            if not isinstance(value, basestring):
+                h.update(str(value))
+            else:
+                h.update(value)
+        return h.hexdigest()
+
+
+class Artifact(Model):
```

Comment:
I know we said we didn't want to do drastic remodeling as part of this conversion, but I'm wondering if it would be good to use this model to hold (or maybe just reference) the various catalog models from Pulp 2.

The LazyCatalogEntry model for lazy sync is the ``file`` (we have that here), ``importer_id``, ``unit_id`` (since this relates to content we don't need that), ``unit_type_id``(``content`` contains the type so we don't need that either), ``url``, checksum + type (we have that here), ``revision`` (do we need that if we have transactions?), and ``data`` (currently unused).

So if we included the URL(s) the artifact came from and the importer(s) that pulled it in, we've duplicated the lazy catalog. If we added a weight to each of those URLs we have the alternate content sources (I think). Does that sound crazy?